[CC 1.0.0] Read APIs from File System [master] (#4968)
wso2/docs-apim: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-apis-as-immutable-gateway.md

> # Deploy APIs as immutable gateway

Reviewer: ```suggestion
# Deploy APIs as Immutable Gateway
```
Can we reword this title so that users can understand it easily? I'm wondering if "immutable" might be too complex information for a title.

Author: Had a discussion with team and decided to move forward with the term Immutable.

> In this case, the APIs will be deployed during the startup. After startup, no APIs can be added using apictl. 

Reviewer: ```suggestion
When deploying APIs as an immutable Gateway, the APIs are deployed during startup. After startup, no APIs can be added using apictl. 
```

For that, the apictl projects needs to be mounted to the `docker-compose/resources/adapter/artifacts/apis` directory 
(default location) of the adapter. 

> 

Reviewer 2: Deploy APIs as an Immutable gateway?

Reviewer 2: In this case, the APIs can be deployed to the gateway in an immutable manner. For that, the apictl projects need to be mounted to the `docker-compose/resources/adapter/artifacts/apis` directory 
(default location) of the adapter. At the startup of the adapter, it will pick the projects. After startup, APIs should be added using apictl.

Author: fixed with https://github.com/wso2/docs-apim/pull/4968/commits/58c025c764f9d7349b1e12c3709188929873dce5

> ## Step 1 - Download apictl and Set the Path Variable 

Reviewer: ```suggestion
## Step 1 - Download apictl and set the path variable 
```

First download [apictl](https://github.com/wso2/product-apim-tooling/releases) locally and extract it into a folder of your choice. Then, add its location to your PATH variable.

```
export PATH=$PATH:<directory-of-the-apictl-executable>
```

## Step 2 - Create an API Project using apictl

Let's create your first project "petstore" using an open API definition. The following `apictl init` command will generate a project folder containing all the necessary files.

!!! warning

>     If you have used a previous version of apictl before, remember to delete the directories `.wso2apictl` and `.wso2apictl.local` that are located in your home directory. Deleting them will make the newer apictl create them again, with content compatible with the current version.

Reviewer: ```suggestion
    If you have used a previous version of apictl before, remember to delete the directories `.wso2apictl` and `.wso2apictl.local` that are located in your home directory. Deleting them will make the newer apictl create them again with content compatible with the current version.
```

``` bash
apictl init petstore --oas https://petstore.swagger.io/v2/swagger.json
```

!!! note

>     Instead of initializing the api project, you can [export the apis from api manager]({{base_path}}/install-and-setup/setup/api-controller/managing-apis-api-products/migrating-apis-to-different-environments.md) as well.

Reviewer: ```suggestion
    Instead of initializing the API project, you can [export the APIs from API Manager]({{base_path}}/install-and-setup/setup/api-controller/managing-apis-api-products/migrating-apis-to-different-environments) as well.
```

## Step 3 - Copy the project to the resources/enforcer/artifacts/apis directory

The created apictl project needs to be copied to `docker-compose/resources/adapter/artifacts/apis` directory. These
apictl projects could be copied either in directory format or in zipped format. 

> And the apictl project could be the api artifact exported from the API Manager too. 

Reviewer: ```suggestion
The apictl project can be the API artifact exported from the API Manager as well. 
```

> The docker-compose file would mount that directory to the `/home/wso2/artifacts/apis`. 

Reviewer: ```suggestion
The docker-compose file mounts that directory to the `/home/wso2/artifacts/apis`. 
```

All the apictl projects in the directory will be deployed during the startup. 

## Step 4 - update config.toml

```
[adapter.server]
    enabled = false
```

!!! note

>     If this configuration is set to true, the mounted api projects will be deployed during the startup. But the it will not have

Reviewer: ```suggestion
    If this configuration is set to true, the mounted API projects are deployed during startup. However, they will not have
```

>     the immutable nature as the APIs could be added after startup using apictl.

Reviewer: ```suggestion
    the immutable nature as the APIs can be added after startup using apictl.
```

!!! note

>     If the control plane is enabled, the mounted api projects will not be deployed.    

Reviewer: ```suggestion
    If the control plane is enabled, the mounted API projects are not deployed.    
```
